Release v0.1.0

pjhartzell opened the pull request:
**Description:**

Initial release.

Generates worldwide COGs for each NetCDF variable, leading to one STAC Item for each NetCDF and corresponding worldwide COGs.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).

pjhartzell committed:
release: v0.1.0

gadomski reviewed (approved):
LGTM. Try pinning mypy to fix the CI issues, e.g. https://github.com/stac-utils/stactools/pull/376/commits/032b0a7aebabbbd8afa873acb0619c06e6d16ba8#diff-2b4945591edfeaa4cf4d3f155e66d4b43d1bda7a55d881d5cf3107f1b05abbbcR8

pjhartzell committed:
fix: lower bound mypy to avoid bug in CI

pjhartzell committed:
fix: pin to python to 3.10.6 in CI to avoid mypy bug

pjhartzell committed:
fix: alternate mypy bug workaround

Move pre-commit repo versions forward rather than pinning to older Python in CI

pjhartzell commented:
Ah. Thanks
